### PR TITLE
[Components with more than 1 service] LPS-185842 Refactor components from Commerce modules

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://issues.liferay.com/browse/LPS-185842

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:commerce

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:commerce

**Comment by Kvale1733** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
@MrJohn1911 No regressions found. Just need to fix the sf issue.

```
     [java] java.lang.Exception: Found 1 formatting issues:
     [java] 1: ./modules/apps/commerce/commerce-discount-api/src/main/java/com/liferay/commerce/discount/target/CommerceDiscountProductTarget.java expected:<...ctTarget \{
     [java] 
     [java]   public [void contributeDocument(
     [java]       Document document, CommerceDiscount commerceDiscount)
     [java]     throws PortalException;
     [java] 
     [java]   public String getKey()];
     [java] 
     [java]   public void post...> but was:<...ctTarget \{
     [java] 
     [java]   public [String getKey();
     [java] 
     [java]   public void contributeDocument(
     [java]       Document document, CommerceDiscount commerceDiscount)
     [java]     throws PortalException];
     [java] 
     [java]   public void post...>
     [java] 
     [java]   at com.liferay.source.formatter.SourceFormatter.format(SourceFormatter.java:466)
     [java]   at com.liferay.source.formatter.SourceFormatter.main(SourceFormatter.java:299)
```

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
Last push fixes sf [here](https://github.com/liferay-core-infra/liferay-portal/pull/4733/commits/020bfaf241f6f3d96863ce81d02ad05d16a40d13).

File: modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/internal/target/BaseCommerceDiscountProductTarget.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import com.liferay.portal.kernel.search.filter.Filter;
+
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Joao Victor Alves
+ */
+public abstract class BaseCommerceDiscountProductTarget
+	implements CommerceDiscountProductTarget {
+
```
Comment (rafaprax) [inline modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/internal/target/BaseCommerceDiscountProductTarget.java:36]: public abstract String getFieldName()

**Comment by MrJohn1911** [comment]
ci:test:commerce

**Comment by Kvale1733** [comment]
ci:test:stable

**Comment by Kvale1733** [comment]
@MrJohn1911 No regressions found. Semver issue only.
```
[exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/commerce/commerce-discount-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.commerce.discount.target               MAJOR      1.1.0      1.1.0      2.0.0      VERSION INCREASE REQUIRED
     [exec]     	-   interface  com.liferay.commerce.discount.target.CommerceDiscountSKUTarget
     [exec]     		-   access     abstract
     [exec]     		-   method     contributeDocument(com.liferay.portal.kernel.search.Document,com.liferay.commerce.discount.model.CommerceDiscount)
     [exec]     			-   access     abstract
     [exec]     			-   return     void
     [exec]     		-   method     postProcessContextBooleanFilter(com.liferay.portal.kernel.search.filter.BooleanFilter,com.liferay.commerce.product.model.CPInstance)
     [exec]     			-   access     abstract
     [exec]     			-   return     void
     [exec]     [Baseline Warning] Bundle Version Change Recommended: 17.0.0
```
[Log](https://testray.liferay.com/reports/production/logs/2023-06/test-1-32/test-portal-acceptance-pullrequest(master)/176/modules-semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
Last push fixes baseline.

File: modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/internal/target/ApplyToProductCommerceDiscountProductTargetImpl.java [hunk 2]
```diff
@@ -14,66 +14,30 @@
...
+		"commerce_discount_target_cp_definition_ids";
 
-		document.addKeyword(
-			"commerce_discount_target_cp_definition_ids",
-			TransformUtil.transformToLongArray(
-				_commerceDiscountRelLocalService.getCommerceDiscountRels(
-					commerceDiscount.getCommerceDiscountId(),
-					CPDefinition.class.getName()),
-				CommerceDiscountRel::getClassPK));
+	public String getFieldName() {
```
Comment (rafaprax) [inline modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/internal/target/ApplyToProductCommerceDiscountProductTargetImpl.java:34]: please add the `@Override` annotation

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:commerce

**Comment by MrJohn1911** [comment]
ci:stop

**Comment by MrJohn1911** [comment]
ci:test:commerce

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:commerce

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:commerce

**Comment by catarineaguiar** [comment]
we are facing some CI issues, so let's wait for the fix to be merged to re-run test suites.

Thanks,
Cat

**Comment by zhangyan0701** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
ci:test:startup

**Comment by zhangyan0701** [comment]
ci:test:commerce

**Comment by zhangyan0701** [comment]
No regressions found.

**Review by dantewang — APPROVED** [review]

**Comment by dantewang** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/136876

**Pull request closed without merging**